packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -310,12 +315,84 @@ export function createChildLogger(props?: {
 	return ChildLogger.create(props?.logger, props?.namespace, props?.properties);
 }
 
+/**
+ * Contains sampling strategies for telemetry.
+ *
+ * @remarks This is an object instead of an enum for better comaptibility
+ * https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums
+ */
+export const SamplingStrategy = {
+	/**
+	 * Samples a subset of data from a larger set by selecting every nth event
+	 */
+	SYSTEMATIC: "Systematic",
+	/**
+	 * Samples a subset of data from a larger set using a controlled psuedo-random chance.
+	 */
+	RANDOM: "Random",
+} as const;
+
+/**
+ * This type should be used as the "Enum" for {@link SamplingStrategy}.
+ */
+export type ESamplingStrategy = typeof SamplingStrategy[keyof typeof SamplingStrategy];
+
+/**
+ * Interface for the configuration for a concrete type of sampling strategy.
+ */
+export interface SamplingConfig {
+	strategy: ESamplingStrategy;
+}
+
+/**
+ * Configuration for systematically sampling telemetry.
+ * See {@link SamplingStrategy.SYSTEMATIC} for more information.
+ */
+export interface SystematicSamplingConfig extends SamplingConfig {
+	strategy: typeof SamplingStrategy.SYSTEMATIC;
+	/**
+	 * Determines whether to send every nth event. E.g. 500 would send 1 out of every 500 events.
+	 */
+	samplingRate: number;
+}
+
+/**
+ * Configuration for randomly sampling telemetry.
+ * See {@link SamplingStrategy.RANDOM} for more information.
+ */
+export interface RandomChanceSamplingConfig extends SamplingConfig {
+	strategy: typeof SamplingStrategy.RANDOM;
+	/**
+	 * Should be a number between 0 and 1.
+	 * The random number generated must be greater than this number for telemetry to be emitted
+	 */
+	percentChance: number;
+}
+
+/**
+ * The state for systematic sampling of an event.
+ * This is necessary to ensure we are keeping a count of total events
+ * so we can sample every nth event correctly.
+ */
+interface SystematicSamplingState {
+	unsampledEventCount: number;
+}
+
+/**
+ * Union type for all different types of Sampling States
+ * @remarks This type leaves us room to continue to add to if/when we add new sampling strategies.
+ */
+type SamplingState = SystematicSamplingState;
+
 /**
  * ChildLogger class contains various helper telemetry methods,
  * encoding in one place schemas for various types of Fluid telemetry events.
  * Creates sub-logger that appends properties to all events
```

Comment:
Probably worth adding a note to the class docs about sampling support. I would recommend putting it and the existing note about creating sub-loggers into a `@remarks` block.